dwave_micro_client_dimod/tiling.py

```diff
@@ -0,0 +1,167 @@
+        self.structure = Structure(sorted(tile.nodes), sorted(tile.edges), tile.adj)
+
+        nodes_per_cell = t * 2
+        edges_per_cell = t * t
+        m = n = int(ceil(sqrt(ceil(len(sampler.structure.nodelist) / nodes_per_cell)))) # assume square lattice shape
+        system = dnx.chimera_graph(m, n, t, node_list=sampler.structure.nodelist, edge_list=sampler.structure.edgelist)
+        c2i = {chimera_index: linear_index for (linear_index, chimera_index) in system.nodes(data='chimera_index')}
+        sub_c2i = {chimera_index: linear_index for (linear_index, chimera_index) in tile.nodes(data='chimera_index')}
+
+        # Count the connections between these qubits
+        def _between(qubits1, qubits2):
+            edges = [edge for edge in system.edges if edge[0] in qubits1 and edge[1] in qubits2]
+            return len(edges)
+
+        # Get the list of qubits in a cell
+        def _cell_qubits(i, j):
+            return [c2i[(i, j, u, k)] for u in range(2) for k in range(t) if (i, j, u, k) in c2i]
+
+        # get a mask of complete cells
+        cells = [[False for _ in range(n)] for _ in range(m)]
+        for i in range(m):
+            for j in range(n):
+                qubits = _cell_qubits(i, j)
+                cells[i][j] = len(qubits) == nodes_per_cell and _between(qubits, qubits) == edges_per_cell
+
+        # List of 'embeddings'
+        self.embeddings = []
+
+        # For each possible chimera cell check if the next few cells are complete
+        for i in range(m + 1 - sub_m):
+            for j in range(n + 1 - sub_n):
+
+                # Check if the sub cells are matched
+                match = all(cells[i + sub_i][j + sub_j] for sub_i in range(sub_m) for sub_j in range(sub_n))
+
+                # Check if there are connections between the cells.
+                for sub_i in range(sub_m):
+                    for sub_j in range(sub_n):
+                        if sub_m > 1 and sub_i < sub_m - 1:
+                            match &= _between(_cell_qubits(i + sub_i, j + sub_j),
+                                              _cell_qubits(i + sub_i + 1, j + sub_j)) == t
+                        if sub_n > 1 and sub_j < sub_n - 1:
+                            match &= _between(_cell_qubits(i + sub_i, j + sub_j),
+                                              _cell_qubits(i + sub_i, j + sub_j + 1)) == t
+
+                if match:
+                    # Pull those cells out into an embedding.
+                    embedding = {}
+                    for sub_i in range(sub_m):
+                        for sub_j in range(sub_n):
+                            cells[i + sub_i][j + sub_j] = False  # Mark cell as matched
+                            for u in range(2):
+                                for k in range(t):
+                                    embedding[sub_c2i[sub_i, sub_j, u, k]] = {c2i[(i + sub_i, j + sub_j, u, k)]}
+
+                    self.embeddings.append(embedding)
```

Comment:
If I give a sampler without Chimera-structure, what is the exception that is raised?

Reply:
Now throws an error when 0 tiled embeddings are found